### ISSUE TITLE
`aria-label `and` aria-labelledby `on name-prohibited roles should have clearer message

### ISSUE THREAD

**Issue by Zidious** [dequelabs/axe-core#3205]
Ref: https://github.com/dequelabs/axe-core-npm/issues/275

As per the ARIA 1.2 spec some roles introduced _Roles which cannot be named (Name prohibited)_ with `aria-label` and `aria-labelledby`. These roles include: 
```
caption code deletion emphasis generic insertion paragraph presentation strong subscript superscript
```
Example provided: [name-prohibited.html](https://joanmarie.github.io/aria/validator-tests/name-prohibited.html). 

Note: it does flag as a violation but message is not clear.

```JS
  <div id="aria-label-1" role="code" aria-label="foo">bar</div>
```
Output: 
```
Fix all of the following: ARIA attribute cannot be used, add a role attribute or use a different element: aria-label
```

```JS
  <code aria-label="foo">bar</code>
```
Output:
```
Fix all of the following: ARIA attribute is not well supported on the element and the text content will be used instead: aria-label
```
